vim/core/oni-core-interop/plugin/init.vim

```diff
@@ -42,6 +52,72 @@ function OniNotifyEvent(eventName)
     call OniNotify(["event", a:eventName, context])
 endfunction
 
+function User_buffers() " help buffers are always unlisted, but quickfix buffers are not
+  return filter(range(1,bufnr('$')),'buflisted(v:val) && "quickfix" !=? getbufvar(v:val, "&buftype")')
+endfunction
+
+function OniGetAllBuffers()
+  let l:buffers = []
+  let l:bufnums = User_buffers()
+  if exists("l:bufnums")
+    for l:bufnum in l:bufnums
+      try
+        let l:buffer = OniGetEachContext(l:bufnum)
+        let l:buffers += [l:buffer]
+      catch /.*/
+      "Probably dont want this outside of a debugging scenario
+      " echohl WarningMsg
+      " echo v:exception
+      " echohl none
+    endtry
+    endfor
+    return l:buffers
+  endif
+endfunction
+
+
+function OniGetEachContext(bufnum)
+  let l:context = {}
+    let l:bufpath = bufname(a:bufnum)
+
+    if strlen(l:bufpath)
+      "TODO: python can be used to retrieve the lines a buffer and len return 
```

Comment:
As mentioned in the comments, unfortunately Python isn't an option for us (because there is no guarantee it's available on Windows). Lua might be an option though.

Reply:
Ah I had it somewhere in my head that neovim was compiled with python by default so anyone who had neovim would have setup up python but that was a fairly random assumption on my part. As I haven't been able to get to grips with it anyway ill take that comment out

Reply:
Ah ya, well it does work out of the box on OSX (and Linux too I believe) so it's an easy assumption to make! 😄